posts/modular_forms.md

```diff
@@ -0,0 +1,123 @@
+---
+author: 'Chris Birkbeck'
+category: 'New in mathlib'
+date: 2022-12-21 12:41:21 UTC+01:00
+description: ''
+has_math: true
+link: ''
+slug: modular-forms
+tags: ''
+title: Modular forms
+type: text
+---
+
+
+In [PR# 13250](https://github.com/leanprover-community/mathlib/pull/13250) 	we define modular forms, cusp forms and prove that they form a complex vector space. These are analytic functions of number theoretic interest with strong links to geometry, representation theory and analysis. Most famously they are a  key ingredient in the proof of Fermat's Last Theorem. In this post we discuss the formalization process, motivate some design choices and map out some future work.
+
+<!-- TEASER_END -->
+
+Before going any further I should mention that this isn't the first time modular forms have been defined in Lean. Back in 2018, for Kevin Buzzard's birthday several people defined modular forms (amongst other things) [here](https://github.com/semorrison/kbb). Although the current definition looks quite different, it was of great use when I started learning Lean. Moreover, the current version benefited immensely from great feedback from many people, including Riccardo Brasca, Kevin Buzzard, David Loeffler, Jireh Loreaux, Heather Macbeth and Eric Wieser.
+
+# Basic definitions
+
+At their most basic, modular forms are functions from the complex upper half plane $\mathbb{H}:=\\{ z \in \mathbb{C} \mid 0 \lt  Im(z)\\}$ to $\mathbb{C}$ satisfying certain properties. Before giving the definition, let's first define an action on this space of functions.
+
+For any 
+
+$$\gamma =
+\left(\begin{array}{cc} 
+a & b\\ 
+c & d
+\end{array}\right)
+$$
+
+in $\mathrm{GL}_2(\mathbb{R})^+$ ($2 \times 2$ matrices with real entries and positive determinant) the weight $k \in \mathbb{Z}$ action of $\gamma$ on $f : \mathbb{H} \to \mathbb{C}$ is given by $$(f \mid_k \gamma) (z):=\mathrm{det} (\gamma)^{k-1} (cz+d)^{-k} f\left ( \frac{az+b}{cz+d}\right ).$$ One easily checks that this defines a right action on this space of functions, known as the weight $k$ *slash action*.
```

Comment:
I would right `\left(` `\right)` without a space (do they not work)?

Reply:
Maybe I'm not understanding what you mean, but i tried removing the space and it wouldnt work.